src/main/java/org/opensearch/security/support/SecurityUtils.java

```diff
@@ -140,4 +149,35 @@ private static String resolveEnvVar(String envVarName, String mode, boolean bc,
             return bc ? Hasher.hash(envVarValue.toCharArray(), settings) : envVarValue;
         }
     }
+
+    public static boolean matchesHostPatterns(WildcardMatcher hostMatcher, InetAddress address, String hostResolverMode) {
+        if (hostMatcher == null || address == null) {
+            return false;
+        }
+        List<String> valuesToCheck = new ArrayList<>(List.of(address.getHostAddress()));
+        if (hostResolverMode != null
+            && (hostResolverMode.equalsIgnoreCase("ip-hostname") || hostResolverMode.equalsIgnoreCase("ip-hostname-lookup"))) {
+            final String hostName = address.getHostName();
+            valuesToCheck.add(hostName);
+        }
+
+        return valuesToCheck.stream().anyMatch(hostMatcher);
+    }
+
+    public static boolean matchesCidrPatterns(ClientBlockRegistry<InetAddress> clientBlockRegistry, InetAddress address) {
+        if (clientBlockRegistry == null || address == null) {
+            return false;
+        }
+        AuthFailureListener authFailureListener = (AuthFailureListener) clientBlockRegistry;
+        String hostAddress = address.getHostAddress();
+        return authFailureListener.getIgnoreHosts().parallelStream().filter(pattern -> pattern.indexOf('/') != -1).anyMatch(pattern -> {
+            try {
+                SubnetInfo subnetInfo = cidrCache.computeIfAbsent(pattern, authFailureListener::getSubnetForCidr);
```

Comment:
Ignore the comment above. Looks like the dynamic config is already being handled 🙌